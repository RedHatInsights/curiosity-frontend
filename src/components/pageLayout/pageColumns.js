import React from 'react';
import PropTypes from 'prop-types';
import { Flex, FlexItem } from '@patternfly/react-core';
import { helpers } from '../../common';

/**
 * Render page columns based on children.
 *
 * @param {object} props
 * @param {Node} props.children
 * @param {string} props.className
 * @returns {Node}
 */
const PageColumns = ({ children, className }) => (
  <Flex spaceItems={{ sm: 'spaceItemsNone' }} className={`curiosity-page-columns ${className}`}>
    {React.Children.toArray(children)
      .filter(child => React.isValidElement(child))
      .map(child => (
        <FlexItem
          key={(child.key && `page-column-${child.key}`) || helpers.generateId()}
          style={{ flexBasis: 0 }}
          grow={{ sm: 'grow' }}
          className="curiosity-page-columns-column"
        >
          {child}
        </FlexItem>
      ))}
  </Flex>
);

/**
 * Prop types.
 *
 * @type {{children: Node, className: string}}
 */
PageColumns.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

/**
 * Default props.
 *
 * @type {{className: string}}
 */
PageColumns.defaultProps = {
  className: ''
};

export { PageColumns as default, PageColumns };
