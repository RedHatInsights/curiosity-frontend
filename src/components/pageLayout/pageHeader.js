import React from 'react';
import PropTypes from 'prop-types';
import {
  PageHeader as RcsPageHeader,
  PageHeaderTitle
} from '@redhat-cloud-services/frontend-components/components/PageHeader';

/**
 * Render a platform page header.
 *
 * @param {object} props
 * @param {Node} props.children
 * @returns {Node}
 */
const PageHeader = ({ children }) => (
  <RcsPageHeader>
    <PageHeaderTitle title={children} />
  </RcsPageHeader>
);

/**
 * Prop types.
 *
 * @type {{children: Node}}
 */
PageHeader.propTypes = {
  children: PropTypes.node.isRequired
};

/**
 * Default props.
 */
PageHeader.defaultProps = {};

export { PageHeader as default, PageHeader };
