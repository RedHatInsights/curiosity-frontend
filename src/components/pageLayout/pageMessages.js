import React from 'react';
import PropTypes from 'prop-types';
import { Section } from '@redhat-cloud-services/frontend-components/components/Section';

/**
 * Render a platform toolbar section.
 *
 * @param {object} props
 * @param {Node} props.children
 * @param {string} props.className
 * @returns {Node}
 */
const PageMessages = ({ children, className, ...props }) => (
  <Section className={`curiosity-page-messages ${className}`} {...props}>
    {children}
  </Section>
);

/**
 * Prop types.
 *
 * @type {{children: Node, className: string}}
 */
PageMessages.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string
};

/**
 * Default props.
 *
 * @type {{className: string}}
 */
PageMessages.defaultProps = {
  className: ''
};

export { PageMessages as default, PageMessages };
