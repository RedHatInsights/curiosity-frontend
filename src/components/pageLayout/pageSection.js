import React from 'react';
import PropTypes from 'prop-types';
import { Section } from '@redhat-cloud-services/frontend-components/components/Section';

/**
 * Render a platform page section.
 *
 * @param {object} props
 * @param {Node} props.children
 * @returns {Node}
 */
const PageSection = ({ children, ...props }) => <Section {...props}>{children}</Section>;

/**
 * Prop types.
 *
 * @type {{children: Node}}
 */
PageSection.propTypes = {
  children: PropTypes.node.isRequired
};

/**
 * Default props.
 */
PageSection.defaultProps = {};

export { PageSection as default, PageSection };
