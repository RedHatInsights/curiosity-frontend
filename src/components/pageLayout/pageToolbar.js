import React from 'react';
import PropTypes from 'prop-types';
import { Section } from '@redhat-cloud-services/frontend-components/components/Section';

/**
 * Render a platform toolbar section.
 *
 * @param {object} props
 * @param {Node} props.children
 * @returns {Node}
 */
const PageToolbar = ({ children, ...props }) => <Section {...props}>{children}</Section>;

/**
 * Prop types.
 *
 * @type {{children: Node}}
 */
PageToolbar.propTypes = {
  children: PropTypes.node.isRequired
};

/**
 * Default props.
 */
PageToolbar.defaultProps = {};

export { PageToolbar as default, PageToolbar };
