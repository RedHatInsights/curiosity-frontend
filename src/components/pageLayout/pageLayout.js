import React from 'react';
import PropTypes from 'prop-types';
import { PageSection as Main } from '@patternfly/react-core';
import { PageHeader } from './pageHeader';
import { PageSection } from './pageSection';
import { PageToolbar } from './pageToolbar';

/**
 * ToDo: Reevaluate, import for Main component from @redhat-cloud-services/frontend-components
 * Fallback towards PF PageSection. Named export for Main is overridden by default connected export.
 */
/**
 * Render a platform page layout.
 *
 * @param {object} props
 * @param {Node} props.children
 * @returns {Node}
 */
const PageLayout = ({ children }) => (
  <React.Fragment>
    {React.Children.toArray(children).filter(child => React.isValidElement(child) && child.type === PageHeader)}
    {React.Children.toArray(children).filter(child => React.isValidElement(child) && child.type === PageToolbar)}
    <Main className="curiosity">
      {React.Children.toArray(children).filter(child => child.type !== PageHeader && child.type !== PageToolbar)}
    </Main>
  </React.Fragment>
);

/**
 * Prop types.
 *
 * @type {{children: Node}}
 */
PageLayout.propTypes = {
  children: PropTypes.node.isRequired
};

/**
 * Default props.
 */
PageLayout.defaultProps = {};

export { PageLayout as default, PageLayout, PageHeader, PageSection, PageToolbar };
