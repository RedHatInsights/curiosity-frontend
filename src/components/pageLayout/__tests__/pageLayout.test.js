import React from 'react';
import { mount } from 'enzyme';
import { PageLayout } from '../pageLayout';

describe('PageLayout Component', () => {
  it('should render a basic component', () => {
    const props = {};

    const component = mount(
      <PageLayout {...props}>
        <span className="test">lorem</span>
      </PageLayout>
    );

    expect(component).toMatchSnapshot('basic');
  });
});
