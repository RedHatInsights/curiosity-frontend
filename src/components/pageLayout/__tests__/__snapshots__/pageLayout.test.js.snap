// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout Component should render a basic component: basic 1`] = `
<React.Fragment>
  <PageSection
    className="curiosity "
    padding={
      {
        "default": "noPadding",
      }
    }
  >
    <span
      className="test"
    >
      lorem
    </span>
  </PageSection>
</React.Fragment>
`;

exports[`PageLayout Component should render header and section children: multiple children 1`] = `
<pagelayout
  props="{
  "children": [
    {
      "displayName": "PageSection",
      "props": {
        "children": "ipsum",
        "className": ""
      }
    },
    {
      "displayName": "PageHeader",
      "props": {
        "children": "lorem",
        "productLabel": null
      }
    }
  ],
  "className": ""
}"
>
  <section
    class="pf-l-page-header pf-c-page-header pf-l-page__main-section pf-c-page__main-section pf-m-light"
    widget-type="InsightsPageHeader"
  >
    <div
      class="pf-l-flex pf-m-justify-content-space-between-on-sm"
    >
      <div
        class=""
      >
        <h1
          class="pf-c-title pf-m-2xl pf-u-mb-sm"
          data-ouia-component-id="OUIA-Generated-Title-1"
          data-ouia-component-type="PF4/Title"
          data-ouia-safe="true"
          widget-type="InsightsPageHeaderTitle"
        >
          lorem
        </h1>
      </div>
    </div>
  </section>
  <section
    class="pf-c-page__main-section pf-m-no-padding curiosity "
  >
    <section
      class="curiosity-page-section "
    >
       
      ipsum
       
    </section>
  </section>
</pagelayout>
`;
